### DISCUSSION TITLE
IntelliSense for Fastify using JS

### DISCUSSION THREAD

**Discussion by bigansh** [fastify/fastify#3605] [Q&A] [answered]
Hey, so I just started using Fastify & I noticed that there are no options/type definitions while typing the code. I think that IntelliSense is not supported in Fastify. How can one solve this? I really want to use Fastify.

**Comment by Eomm** [comment, marked as answer]
To turn it on you need to require it like this:

```
const { fastify } = require('fastify')
```

or use ESM `import`

**Reply by bigansh** [reply]
For using them in `Routes` with parameters like (fastify, options, done), what JSDoc settings do I need to use?

**Reply by mcollina** [reply]
https://www.fastify.io/docs/latest/Reference/TypeScript/#code-completion-in-vanilla-javascript
